### PR TITLE
various changes

### PR DESCRIPTION
**Pull request by DavidBJaffe**
- add umi ratio filtering
- reorganize and beautify summary
- add some cpu stats (via non-public arguments)
- some operations now much faster
- add ability to use analysis set ids for TCR, BCR, GEX arguments (preceded by S)
- code cleanup
  * simplify some command line argument processing
  * remove unneeded stuff from Cargo.toml files
  * split out crate enclone_print to reduce compilation time
- testing changes
  * add a speed test that you can run by typing "speed"
  * make an error message more informative

### CODE REVIEW

File: enclone/src/proc_args3.rs [hunk 1]
```diff
@@ -59,6 +59,99 @@ fn expand_integer_ranges(x: &str) -> String {
...
+            let o = Command::new("curl")
+                .arg(url)
+                .output()
+                .expect("failed to execute xena http");
+            let m = String::from_utf8(o.stdout).unwrap();
+            if m.contains("502 Bad Gateway") {
+                eprintln!(
+                    "\nWell this is sad.  The URL \
+                    http://xena/api/analysis_sets/{} yielded a 502 Bad Geteway \
+                    message.  Either try again later or ask someone for help.\n\n",
```
Comment (wyattmcdonnell) [inline enclone/src/proc_args3.rs:93]:
```suggestion
                    "\nWell, this is sad.  The URL \
                    http://xena/api/analysis_sets/{} returned a 502 Bad Gateway \
                    message.  Please try again later or ask someone for help.\n\n",
```

File: enclone/src/proc_args3.rs [hunk 2]
```diff
@@ -59,6 +59,99 @@ fn expand_integer_ranges(x: &str) -> String {
...
+                let mut ids = m.between("\"analysis_ids\":[", "]").to_string();
+                ids = ids.replace(" ", "");
+                ids = ids.replace("\n", "");
+                let ids = ids.split(',').collect::<Vec<&str>>();
+                let mut ids2 = Vec::<String>::new();
+
+                // Remove wiped analysis ids.
+
+                for j in 0..ids.len() {
+                    let url = format!("https://xena.fuzzplex.com/api/analyses/{}", ids[j]);
```
Comment (wyattmcdonnell) [inline enclone/src/proc_args3.rs:109]: Given our new architecture can this be migrated to a different/non-`fuzzplex` URL?

File: enclone/src/proc_args3.rs [hunk 3]
```diff
@@ -59,6 +59,99 @@ fn expand_integer_ranges(x: &str) -> String {
...
+                    let o = Command::new("curl")
+                        .arg(url)
+                        .output()
+                        .expect("failed to execute xena http");
+                    let m = String::from_utf8(o.stdout).unwrap();
+                    if m.contains("502 Bad Gateway") {
+                        eprintln!(
+                            "\nWell this is sad.  The URL \
+                            http://xena/api/analyses/{} yielded a 502 Bad Geteway \
+                            message.  Either try again later or ask someone for help.\n",
```
Comment (wyattmcdonnell) [inline enclone/src/proc_args3.rs:119]:
```suggestion
                            "\nWell, this is sad.  The URL \
                            http://xena/api/analyses/{} returned a 502 Bad Gateway \
                            message.  Either try again later or ask someone for help.\n",
```

File: enclone_print/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+[package]
+name = "enclone_print"
+version = "0.2.82"
```
Comment (wyattmcdonnell) [inline enclone_print/Cargo.toml:3]:
```suggestion
version = "0.3.0"
```

File: enclone/src/proc_args3.rs (hunk 2 above)
Comment (DavidBJaffe) [reply enclone/src/proc_args3.rs:109]: Yup, fixing and will push change.

**Pull request merged**